### PR TITLE
Fix(Scripts/BRD) - Magmus, Emperor, Priestess, Arena improvements

### PR DESCRIPTION
**Pull request by patou01**
## Changes Proposed:
Apologies for the huge PR, will try to figure out a way to make following ones smaller.

- Updated Moira Bronzebeard to bossAI. She now casts all her spells and emotes
- Updated Moira Bronzebeard. She is replaced by High priestess of Thaurissan if everybody has saved the princess in a previous run. High Priestess uses her yell. 
- Added High Priestess yell to database
- Updated emperor to BossAI.
- Emperor no longer casts hand of thaurissan at victim but randomly
- Updated emperor to do his yells.
- After arena, spectators become neutral (please verify faction I used)
- Arena bosses are now actually random. Before we were using rand() without a seed and you seemingly always got small spiders and big spider boss
- Magmus is now casting his warstomp from 100% hp. Rewrote to BossAI. ~Correctly yells after opening doors~ was outside.
- Fire statues at Magmus now start on engage. They have random grouping and more random in their cast.
- Flame keepers are now pooled. Only maximum 2 of them per dungeon at any time

Edit with additional fixes:
- Added scripts for arena bosses
- Priestess and Moira are now mages and not paladins.
- reworked the Lyceum event a bit too, since the data for the dungeon was not set properly. Doors are opened by the instance and not by the braziers anymore (fundamentally, it's the same for the players, but I needed setData to work for Moira/Priestess swap which is done after the lyceum event to have Moira spawned).

## Issues Addressed:

- closes https://github.com/azerothcore/azerothcore-wotlk/issues/2689
- closes https://github.com/azerothcore/azerothcore-wotlk/issues/1774 (was mainly closed already, now just some cosmetics and logic)
- closes https://github.com/azerothcore/azerothcore-wotlk/issues/1779
- closes https://github.com/azerothcore/azerothcore-wotlk/issues/7726

## SOURCE:

- flamekeepers: https://wowpedia.fandom.com/wiki/Lyceum
- other stuff can be seen here in classic https://youtu.be/hnRIUMTjygU?t=12035 but is common sense
- Priestess yell https://wowpedia.fandom.com/wiki/High_Priestess_of_Thaurissan 

## Tests Performed:

- builds on my machine with VS 2019 on win10.
- poked around on fights

## How to Test the Changes:

0. remove GM tag before entering, or you will not get the priestess
1. Go to arena, you should be able to see something else than borers and a big blue spider at the end.
2. Finish arena, mobs upstairs should be neutral.
3. Go to lyceum, should be only 2 flame keepers
4. Fire statues should be off until engaging Magmus
5. Go to emperor room and start killing mobs, yells should happen. Pull the boss, he should have correct yells. It's good to have several people here to confirm that the random cast is done correctly.
6. Finish quest 4004 or 4363 and start again (reset dungeon), Moira should be gone and the priestess took her place. Note that this happens when the doors to Magmus open, so you need to finish the lyceum event.

7. Try all arena bosses 1 by 1 (probably just .add them) to verify the spells

## Known Issues and TODO List:

- [ ] Spells need to be defined before merging. Valid for all arena bosses and Moira/Priestess
- [ ] Arena waves need to be confirmed, how many mobs of each should spawn.
- [ ] Timers need to be defined
- [ ] The spawnpoints and waypoints of the flame keepers need to be reviewed and changed. Should be 8 keepers
- [ ] If Emperor has done 5 yells for the senators (ie all of them), he will not yell "Come to aid the throne" if a mob is still alive in the room. This is an edge case that I couldn't find an easy "GetCreaturesWithinRange()" to fix so I didn't bother.

## Credit
Thanks to https://github.com/Asthalor for the sql request for the yells for Emperor.

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by patou01** [comment]
I copied the for loop where the CI fails from another part of the code. I can't figure it out in a few minutes, maybe someone knows the issue?
````
fatal error: member access into incomplete type 'const MapReference'
                    if (Player* player = itr->GetSource())
                                            ^
/home/runner/work/azerothcore-wotlk/azerothcore-wotlk/src/server/game/Maps/MapRefMgr.h:12:7: note: forward declaration of 'MapReference'
````
code:
````
 Map::PlayerList const& lPlayers = instance->GetPlayers();
MoiraSaved = true;
if (!lPlayers.isEmpty())
{
    for (Map::PlayerList::const_iterator itr = lPlayers.begin(); itr != lPlayers.end(); ++itr)
    {
        if (Player* player = itr->GetSource())
        {
            bool hasSavedPrincess = (player->GetQuestStatus(PRINCESS_QUEST_HORDE) == QUEST_STATUS_REWARDED)
                                        || (player->GetQuestStatus(PRINCESS_QUEST_ALLIANCE) == QUEST_STATUS_REWARDED);
            if (!hasSavedPrincess)
            {
                MoiraSaved = false;
            }
        }
    }
} 
````

**Comment by Nyeriah** [comment]
> I copied the for loop where the CI fails from another part of the code. I can't figure it out in a few minutes, maybe someone knows the issue?
> 
> ```
> fatal error: member access into incomplete type 'const MapReference'
>                     if (Player* player = itr->GetSource())
>                                             ^
> /home/runner/work/azerothcore-wotlk/azerothcore-wotlk/src/server/game/Maps/MapRefMgr.h:12:7: note: forward declaration of 'MapReference'
> ```
> 
> code:
> 
> ```
>  Map::PlayerList const& lPlayers = instance->GetPlayers();
> MoiraSaved = true;
> if (!lPlayers.isEmpty())
> {
>     for (Map::PlayerList::const_iterator itr = lPlayers.begin(); itr != lPlayers.end(); ++itr)
>     {
>         if (Player* player = itr->GetSource())
>         {
>             bool hasSavedPrincess = (player->GetQuestStatus(PRINCESS_QUEST_HORDE) == QUEST_STATUS_REWARDED)
>                                         || (player->GetQuestStatus(PRINCESS_QUEST_ALLIANCE) == QUEST_STATUS_REWARDED);
>             if (!hasSavedPrincess)
>             {
>                 MoiraSaved = false;
>             }
>         }
>     }
> } 
> ```

You're missing an include. I suppose that file is missing Player.h

**Comment by patou01** [comment]
I found a logical error in OnPlayerEnter regarding Moira and Priestess. If the first player that enters the dungeon has saved Moira, it'll spawn Priestess. I'll fix that later this week.

File: src/server/scripts/EasternKingdoms/BlackrockMountain/BlackrockDepths/blackrock_depths.cpp [hunk 1]
```diff
@@ -12,6 +12,11 @@
 #include "ScriptMgr.h"
 #include "WorldSession.h"
 
+#define TIMER_IRONHAND_FLAMES 16000
+#define TIMER_IRONHAND_RAND_PART 3000
+#define N_IRONHAND_GROUPS 3
+#define SPELL_GOUT_OF_FLAME 15529
```
Comment (Kitzunu) [inline src/server/scripts/EasternKingdoms/BlackrockMountain/BlackrockDepths/blackrock_depths.cpp:18]: please dont use defines. https://www.azerothcore.org/wiki/cpp-code-standards#enumerations-vs-define

**Comment by patou01** [comment]
@kitzunu, your comment should be addressed now. Would you mind reviewing again? Namely one part where I'm not too happy is the spawning of Moira or Priestess.

- I cannot find what should happen if a player that hasn't saved her joins the group once she's in sight, so I took the "logical" way that he shouldn't be allowed to see the priestess and swap her out regardless
- I don't like having the swap called at the end of Lyceum, but using only OnPlayerEntry causes issues that the mob is not in the dungeon before they get in the area of the Lyceum. It could be possible to edit the list of NPCs to spawn, but haven't seen how to do that. Using OnCreatureCreate() doesn't work either (at least not with my code) because AddToWorld() is called after OnCreatureCreate() and therefore I can't unsummon her. I could hack the spawn ID of the unit but that sounds worse than what I have now.

**Comment by patou01** [comment]
Right now this would introduce some slightly annoying bug as Gorosh is bugged, but please review anyway and we can always add a hotfix on top to remove the bugged spell from Gorosh (I'm fixing the whole dungeon atm).

**Comment by patou01** [comment]
@FrancescoBorzi Yes I know, I've got a bit of a pickle due to how it's grown and then someone pushed a fix that I had in here, so I took the whole thing on a different branch and will make a bigger update once I'm happy with the state of what I have (probably within the next 7 days).

I had put "WIP" and removed "ready for review" so that it wouldn't be in the way for people. Is there some way of working that I didn't understand?

**Comment by FrancescoBorzi** [comment]
> I had put "WIP" and removed "ready for review" so that it wouldn't be in the way for people.

that's perfect, no worries

**Comment by patou01** [comment]
@FrancescoBorzi  I think there shouldn't be a merge conflict anymore and it should be okay for review.

**Comment by patou01** [comment]
Help wanted: 
- spell IDs for Moira
- Spell IDs for the arena bosses and timers
- wave compositions for the arena, they depend on the mobs that are spawned but I can't find a source indicating how many of which. For instance if the mobs for the wave are beetles, then there's eg 5 beetles, but the thunder lizards are only 2 because they're stronger.

**Comment by Nyeriah** [comment]
@patou01 do you think this is mergeable as it is and the work here could be continued in another PR(s)?

**Review by Nyeriah — APPROVED** [review]

**Comment by patou01** [comment]
I think it's kinda untested by anyone else than me. Lots of things need adjustement (spells, timers, waves for arena). Once that input is there it's a matter of hours to finalize. But until then it could be problematic, as one arena boss **slaps** (the orc). So I feel something is off in that.

File: src/server/scripts/EasternKingdoms/BlackrockMountain/BlackrockDepths/blackrock_depths.cpp [hunk 2]
```diff
@@ -119,13 +153,18 @@ class at_ring_of_law : public AreaTriggerScript
     {
         if (InstanceScript* instance = player->GetInstanceScript())
         {
+            time_t now = time(nullptr);
```
Comment (Nyeriah) [inline src/server/scripts/EasternKingdoms/BlackrockMountain/BlackrockDepths/blackrock_depths.cpp:156]: what is this being used for?

File: src/server/scripts/EasternKingdoms/BlackrockMountain/BlackrockDepths/blackrock_depths.cpp [hunk 3]
```diff
@@ -46,68 +48,100 @@ class go_shadowforge_brazier : public GameObjectScript
...
+        }
 
-                if (Creature* magmus = ObjectAccessor::GetCreature(*go, instance->GetGuidData(DATA_MAGMUS)))
+        void UpdateAI(uint32 diff) override
+        {
+            events.Update(diff);
+
+            if (flames_enabled)
+            {
+                if (me->HasUnitState(UNIT_STATE_CASTING))
```
Comment (Nyeriah) [inline src/server/scripts/EasternKingdoms/BlackrockMountain/BlackrockDepths/blackrock_depths.cpp:101]: brackets missing here

File: src/server/scripts/EasternKingdoms/BlackrockMountain/BlackrockDepths/boss_emperor_dagran_thaurissan.cpp [hunk 4]
```diff
@@ -86,33 +103,28 @@ class boss_emperor_dagran_thaurissan : public CreatureScript
...
-                HandOfThaurissan_Timer = 5000;
-                //Counter = 0;
-                //}
-            }
-            else HandOfThaurissan_Timer -= diff;
+            events.Update(diff);
 
-            //AvatarOfFlame_Timer
-            if (AvatarOfFlame_Timer <= diff)
+            if (me->HasUnitState(UNIT_STATE_CASTING))
```
Comment (Nyeriah) [inline src/server/scripts/EasternKingdoms/BlackrockMountain/BlackrockDepths/boss_emperor_dagran_thaurissan.cpp:108]: brackets

File: src/server/scripts/EasternKingdoms/BlackrockMountain/BlackrockDepths/boss_eviscerator.cpp [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+                DoCast(SPELL_SHIELD);
+                SpellShieldReady = false;
+                events.ScheduleEvent(SPELL_SHIELD, TIMER_SHIELD);
+            }
+        }
+
+        void UpdateAI(uint32 diff) override
+        {
+            //Return since we have no target
+            if (!UpdateVictim())
```
Comment (Nyeriah) [inline src/server/scripts/EasternKingdoms/BlackrockMountain/BlackrockDepths/boss_eviscerator.cpp:73]: brackets

File: src/server/scripts/EasternKingdoms/BlackrockMountain/BlackrockDepths/boss_gorosh_the_dervish.cpp [hunk 6]
```diff
@@ -35,45 +43,60 @@ class boss_gorosh_the_dervish : public CreatureScript
...
-            {
-                DoCast(me, SPELL_WHIRLWIND);
-                WhirlWind_Timer = 15000;
-            }
-            else WhirlWind_Timer -= diff;
+            events.Update(diff);
 
-            //MortalStrike_Timer
-            if (MortalStrike_Timer <= diff)
+            if (me->HasUnitState(UNIT_STATE_CASTING))
```
Comment (Nyeriah) [inline src/server/scripts/EasternKingdoms/BlackrockMountain/BlackrockDepths/boss_gorosh_the_dervish.cpp:68]: brackets

File: src/server/scripts/EasternKingdoms/BlackrockMountain/BlackrockDepths/boss_grizzle.cpp [hunk 7]
```diff
@@ -36,48 +42,53 @@ class boss_grizzle : public CreatureScript
...
+            events.Update(diff);
 
-            //GroundTremor_Timer
-            if (GroundTremor_Timer <= diff)
-            {
-                DoCastVictim(SPELL_GROUNDTREMOR);
-                GroundTremor_Timer = 8000;
-            }
-            else GroundTremor_Timer -= diff;
+            if (me->HasUnitState(UNIT_STATE_CASTING))
```
Comment (Nyeriah) [inline src/server/scripts/EasternKingdoms/BlackrockMountain/BlackrockDepths/boss_grizzle.cpp:64]: brackets here too

**Comment by Nyeriah** [comment]
> I think it's kinda untested by anyone else than me. Lots of things need adjustement (spells, timers, waves for arena). Once that input is there it's a matter of hours to finalize. But until then it could be problematic, as one arena boss **slaps** (the orc). So I feel something is off in that.

Yes, I understand, that's the problem with PRs that address too many things, they're harder to maintain and to find people to test everything, reason why I wanted to get this merged so we could address the remaining issues

I did the usual bracket-spotting, please recheck the files for missing brackets

Thanks for your time

File: src/server/scripts/EasternKingdoms/BlackrockMountain/BlackrockDepths/blackrock_depths.cpp (hunk 2 above)
Comment (patou01) [reply src/server/scripts/EasternKingdoms/BlackrockMountain/BlackrockDepths/blackrock_depths.cpp:156]: Warding off so it gives 2 minutes between each wipe (yeah somewhat in magic numbers) on line 161. There's probably a better way than that. I couldn't figure if the event today had any kind of reset.

**Comment by patou01** [comment]
Yup, I'll go through it again.

The issue with splitting PRs within a dungeon is that it's all intertwined, if you remember that your patch for the gate from Magmus broke it for me... So it just gets too easy to break everything, and I expect it saves time for the testers to just run once.

An example also of how it grows is that just fixing the arena means you have to add 5 bosses... Unless they're scripted somewhere that I didn't find. 

So not ideal I agree, I try to make them smaller, but this one just blew up, since as soon as I touched something it broke elsewhere.

**Comment by Nyeriah** [comment]
Okay, that's awesome! Thank you. Just try to keep them in a reasonable size, so they can be reviewed, tested and merged within 2-3 days. I think it's probably best if we aim at making PRs that are relatively short so we can process them quickly :)

File: src/server/scripts/EasternKingdoms/BlackrockMountain/BlackrockDepths/blackrock_depths.cpp [hunk 8]
```diff
@@ -46,68 +48,101 @@ class go_shadowforge_brazier : public GameObjectScript
...
+    }
+
+   struct ironhand_guardianAI : public CreatureAI
+    {
+        ironhand_guardianAI(Creature* creature) : CreatureAI(creature) {}
+        bool flames_enabled = false;
+
+        void SetData(uint32 id, uint32 value) override
+        {
+            if (id  == 0)
```
Comment (claudiodfc) [inline src/server/scripts/EasternKingdoms/BlackrockMountain/BlackrockDepths/blackrock_depths.cpp:85]: `if (!id)`

**Comment by claudiodfc** [comment]
### Test results:

- ❌ _Updated Moira Bronzebeard to bossAI. She now casts all her spells and emotes._ **Moira goes oom after 5 casts. is she using the correct spell ids? After Emperor's demise, she goes back to the throne's side casting emote cry but not moving her legs. Are we using the correct emote?**.
- ❌ _Updated Moira Bronzebeard. She is replaced by High priestess of Thaurissan if everybody has saved the princess in a previous run. High Priestess uses her yell._ **Completed the Thrall quest for the chain, reseted instance and used command .go c id 9019 (Emperor) and Moira is still standing next to him.**
- ✔ _Emperor no longer casts hand of thaurissan at victim but randomly_
- ✔ _Updated emperor to do his yells._
- ✔ _After arena, spectators become neutral (please verify faction I used)_
- ❌ _Arena bosses are now actually random. Before we were using rand() without a seed and you seemingly always got small spiders and big spider boss_ **I got lizards + Anub'shiah 1st try. Is Anub'shiah always gonna be the 1st?**
- ✔ _Magmus is now casting his warstomp from 100% hp. Rewrote to BossAI. Correctly yells after opening doors was outside._
- ✔ _Fire statues at Magmus now start on engage. They have random grouping and more random in their cast._
- ✔ _Flame keepers are now pooled. Only maximum 2 of them per dungeon at any time_
- ✔ _Added scripts for arena bosses_
- ✔ _Priestess and Moira are now mages and not paladins._
- ✔ _reworked the Lyceum event a bit too, since the data for the dungeon was not set properly. Doors are opened by the instance and not by the braziers anymore (fundamentally, it's the same for the players, but I needed setData to work for Moira/Priestess swap which is done after the lyceum event to have Moira spawned)._

**Comment by patou01** [comment]
Moira gets replaced after the Lyceum event, so you need to get 2 shadowforge torches to light the braziers. It's a technical/skill limitation (I couldn't figure out a way to prevent her spawn, so I'm swapping her at a moment I know she's there but the users don't usually see her).

The spells are wrong, I'll try to do a check this week-end if I can figure them out, but I don't really know where to start on that.

I feel the first time one spawns the boss after a server restart we get a spider with current implementation. Even after that first spawn however it felt off. I had a feeling that wasn't the case anymore after this PR, I'll re-check that again.

**Comment by claudiodfc** [comment]
I'll check the lyceum event then and look on vmangos for the moira spells

**Comment by claudiodfc** [comment]
I was looking into vmangos core and their spells aren't very far away from what we got. However, you can check their timers since It's the best we have:
https://github.com/vmangos/core/blob/development/src/scripts/eastern_kingdoms/burning_steppes/blackrock_depths/boss_emperor_dagran_thaurissan.cpp#L191

```` c++
enum ePrincess
{
    SPELL_HEAL                  = 15586,
    SPELL_RENEW                 = 10929,
    SPELL_SHIELD                = 10901,
    SPELL_MINDBLAST             = 15587,
    SPELL_SHADOWWORDPAIN        = 15654,
    SPELL_SMITE                 = 10934,
    SPELL_SHADOW_BOLT           = 15537,
    SPELL_OPEN_PORTAL           = 13912
};

**Comment by patou01** [comment]
Yeah might make sense to follow them on there. I'm a bit at a loss at the "spell_portal" for moira. Any idea what it is? Apparently something she should cast on reaching home after the emperor died? If so I'll add it to the to-do list for after this PR.

It could also be that her mana is simply completely wrong?

**Comment by claudiodfc** [comment]
There's some stats that are clearly not right. Also here's an update on spells. Apparently she only casts 3 spells really and it coincides with the information on this wiki page:

https://wowpedia.fandom.com/wiki/Moira_Thaurissan

That should help figuring out why she goes oom so fast in the current `master`.

## Azerothcore
Health | Mana | Level
--- | --- | ---
13592 | 6351 | 55

## Classic vmangos sniffs
Health | Mana | Level
--- | --- | ---
14931 | 6867 | 58

### Extra notes:
These are the spells that she casts:

| caster_id | caster_type | spell_id | Comment
| ---: | --- | ---: | ---:|
| 8929 | Creature | 1604 | Dazed (Irrelevant)
| 8929 | Creature | 8362 | Renew
| 8929 | Creature | 15586 | Heal
| 8929 | Creature | 15587 | Mind Blast

**Comment by claudiodfc** [comment]
The same stats that **Moira Bronzebeard** has should be updated for **High Priestess of Thaurissan**.

**Comment by claudiodfc** [comment]
I went ahead and checked on sniffs for **Emperor Dragan Thaurissan** and found out that our stats are also wrong:

#### Classic sniffs
- HP: 26649
- Level: 58

#### Azerothcore master
- HP: 24291
- Level: 55

#### Correct spell ids

| caster_id | caster_type | spell_id | comment |
| ---: | --- | ---: | ---: |
| 9019 | Creature | 15636 | Avatar of Flame |
| 9019 | Creature | 15658 | Wake up Princess |
| 9019 | Creature | 15713 | Boomerang |
| 9019 | Creature | 17492 | Hand of Thaurissan |

#### Notes
Although Emperor's spells are pretty troll, they are dummy spells for scripts to start and should be considered for scripting. Of course it doesn't really count for this PR but I'm leaving it here in case someone feels interested in scripting them later.

**Comment by patou01** [comment]
I'm curious what this "wake up princess" is, is it supposed to remove CC from the princess or something? I'll fix the timers tomorrow and update the to-do list for after this PR or make new issues...

**Comment by claudiodfc** [comment]
I honestly can't imagine why would they need this spell other than when you engage Emperor Thaurissan. Maybe on vanilla they weren't formation? Also that boomerang spell looks weird too.
If possible try to edit their levels and stats. Thanks!

**Comment by patou01** [comment]
I've done some updates to the spells, it's a bit odd that everybody would have them wrong however, but I don't know what to say. The data from tbc.wowhead suggests 5 spells, https://tbc.wowhead.com/npc=8929/princess-moira-bronzebeard#abilities , but I'm not sure that's actual data or just a guess. The classic one has some 180 odd spells which is what caused the mess in the first place. The spells seem to be in the right range of ids however on that page.

**Comment by claudiodfc** [comment]
Mostly because wowhead isn't sniffed data and that's the main source of information that people go for, so it's usually bugged. VMangos uses at all times sniffed data if possible, so you can expect real data from them. Also wowhead has, not only possible spells but auras and other triggers by those spells, and sometimes, spells like "Wake up princess" don't show up because they weren't "visible". This is why sniffs are so important for quality.

**Comment by patou01** [comment]
But if vmangos sniffed those spells, shouldn't we have the same? I just removed a couple as you said it uses only 3

<html><body>


8929 | Creature | 8362 | Renew
-- | -- | -- | --
8929 | Creature | 15586 | Heal
8929 | Creature | 15587 | Mind Blast


</body>
</html>

Or is vmangos not up to date there?

**Comment by T1ti** [comment]
> I went ahead and checked on sniffs for **Emperor Dragan Thaurissan** and found out that our stats are also wrong:
> 
> #### Classic sniffs
> * HP: 26649
> * Level: 58
> 
> #### Azerothcore master
> * HP: 24291
> * Level: 55
> 
> #### Correct spell ids
> caster_id	caster_type	spell_id	comment
> 9019	Creature	15636	Avatar of Flame
> 9019	Creature	15658	Wake up Princess
> 9019	Creature	15713	Boomerang
> 9019	Creature	17492	Hand of Thaurissan
> #### Notes
> Although Emperor's spells are pretty troll, they are dummy spells for scripts to start and should be considered for scripting. Of course it doesn't really count for this PR but I'm leaving it here in case someone feels interested in scripting them later.

WOTLK modified the level of several dungeons and bosses for the dungeon finder.

**Comment by T1ti** [comment]
> I've done some updates to the spells, it's a bit odd that everybody would have them wrong however, but I don't know what to say. The data from tbc.wowhead suggests 5 spells, https://tbc.wowhead.com/npc=8929/princess-moira-bronzebeard#abilities , but I'm not sure that's actual data or just a guess. The classic one has some 180 odd spells which is what caused the mess in the first place. The spells seem to be in the right range of ids however on that page.

it's not odd at all, all emulators come from good old mangos and have very old data unless updated

**Comment by T1ti** [comment]
> Mostly because wowhead isn't sniffed data and that's the main source of information that people go for, so it's usually bugged. VMangos uses at all times sniffed data if possible, so you can expect real data from them. Also wowhead has, not only possible spells but auras and other triggers by those spells, and sometimes, spells like "Wake up princess" don't show up because they weren't "visible". This is why sniffs are so important for quality.

sniffs are a reference, vmangos is not, unless it has been updated. not ALL the vmangos data is up to date from sniff.

**Comment by claudiodfc** [comment]
Stop contradicting yourself then, sometimes you say vmangos is good for data, sometimes you say it isn't. By this logic sniffing classic is a waste of time because, Blizzard might have changed the difficulty of nearly everything, which they did. So getting sniffs from WOTLK back in the day is impossible.

**Comment by T1ti** [comment]
It's about using logic, the level of the enemies has changed but that doesn't mean the spells they cast are different(very likely it's not, they just retuned it, they didn't rework boss encoutners). So stats from classic sniffs are not a reference, but spell casts probably are.
And anything from Vanilla should always be double checked

**Comment by T1ti** [comment]
https://youtu.be/pBaLSl0wW0M?t=2182
image

Here you can see the High priestess casting shadow bolt and renew

**Comment by T1ti** [comment]
Here's a pretty long fight here https://youtu.be/6GjNiicPEEo?t=20891
So basically Pricness only use those 3 spells (renew/mind blast/heal)

But the High Priestess uses Shadow bolt. This spell list looks accurate https://www.wowdb.com/npcs/10076-high-priestess-of-thaurissan#related:casts . She's so rare that we won't get really more info on her

**Comment by patou01** [comment]
Thanks, I'll go through those videos to see if there's anything I missed. I also don't see her scripted separately in vmangos, so I guess they didn't do that.

It's fun to see they have basically the same implementation as I did, despite me not having looked at their source while developping the code.

**Comment by patou01** [comment]
> Here's a pretty long fight here https://youtu.be/6GjNiicPEEo?t=20891 So basically Pricness only use those 3 spells (renew/mind blast/heal)
> 
> But the High Priestess uses Shadow bolt. This spell list looks accurate https://www.wowdb.com/npcs/10076-high-priestess-of-thaurissan#related:casts . She's so rare that we won't get really more info on her

Eh... So just so I understand that correctly.
Moira: **Renew**, **heal**, mind blast
Priestess: **Renew**, **heal**, shadow bolt, shadow word pain?
(bold being the common part)

**Comment by acidmanifesto** [comment]
@patou01 what is the current status of this pr?

**Comment by patou01** [comment]
> 

Well I'm still not sure of the spells they should have, and after that I think it's good to go...

**Comment by claudiodfc** [comment]
They are the ones I got from the sniffs bro

**Comment by patou01** [comment]
So I've updated it to be as follows:
Moira: Renew, heal, mind blast
Priestess: Renew, heal, shadow bolt, shadow word pain

The timers are probably off but that can be fixed later.

**Pull request merged**